src/graph_ir.rs

```diff
@@ -211,75 +232,57 @@ impl GraphIR {
         let mut output_buffers = vec![];
 
         // Collect the inputs to the algorithm, the incoming/outgoing edges of this node.
-        let edges = &self.adjacent[&node.id];
-        let inputs = node.inputs.iter().map(|p| (p, true));
-        let outputs = node.outputs.iter().map(|p| (p, false));
+        let adjacent_edges = &self.adjacent[&node.id];
+
+        let mut buffers_to_release: Vec<Rc<BufferRef>> =
```

Comment:
What is the `Rc` needed for?

Reply:
Because that's the type returned by this method: <https://github.com/MeadowlarkDAW/audio-graph/blob/main/src/buffer_allocator.rs#L89>

Releasing these buffers immediately will actually allow the allocator to assign the same buffer multiple times to the same node, which is a bug. Instead we want to release them *after* we have finished assigning all buffers to the node.

Reply:
👍